org-roam-graph.el

```diff
@@ -219,19 +219,23 @@ into a digraph."
 
 (defun org-roam-graph--build (&optional node-query)
   "Generate a graph showing the relations between nodes in NODE-QUERY."
-  (unless org-roam-graph-executable
-    (user-error "Can't find %s executable.  Please check if it is in your path"
-                org-roam-graph-executable))
-  (let* ((node-query (or node-query
-                         `[:select [file titles]
-                           :from titles
-                           ,@(org-roam-graph--expand-matcher 'file t)]))
-         (graph      (org-roam-graph--dot node-query))
-         (temp-dot   (make-temp-file "graph." nil ".dot" graph))
-         (temp-graph (make-temp-file "graph." nil ".svg")))
-    (call-process org-roam-graph-executable nil 0 nil
-                  temp-dot "-Tsvg" "-o" temp-graph)
-    temp-graph))
+  (let* ((name org-roam-graph-executable)
+         (exec (ignore-errors (executable-find name))))
+    (unless exec
+      (user-error (concat "Cannot find executable "
+                          (when name
+                            (format "\"%s\" " name))
+                          "to generate the graph.  "
+                          "Please adjust `org-roam-graph-executable'")))
+    (let* ((node-query (or node-query
+                           `[:select [file titles]
+                             :from titles
+                             ,@(org-roam-graph--expand-matcher 'file t)]))
+           (graph (org-roam-graph--dot node-query))
+           (temp-dot (make-temp-file "graph." nil ".dot" graph))
+           (temp-graph (make-temp-file "graph." nil ".svg")))
+      (call-process exec nil 0 nil temp-dot "-Tsvg" "-o" temp-graph)
+      temp-graph)))
```

Comment:
```suggestion
  (unless (stringp org-roam-graph-executable)
    (user-error "`org-roam-graph-executable' must be a string."))
  (unless (executable-find org-roam-graph-executable)
    (user-error "Cannot find executable \"%s\" to generate the graph.
Please adjust `org-roam-graph-executable'"
                org-roam-graph-executable))
  (let* ((node-query (or node-query
                         `[:select [file titles]
                                   :from titles
                                   ,@(org-roam-graph--expand-matcher 'file t)]))
         (graph      (org-roam-graph--dot node-query))
         (temp-dot   (make-temp-file "graph." nil ".dot" graph))
         (temp-graph (make-temp-file "graph." nil ".svg")))
    (call-process exec nil 0 nil temp-dot "-Tsvg" "-o" temp-graph)
    temp-graph))
```
This will cover cases where `org-roam-graph-executable` is the wrong type.
It will also cover cases where the executable is not found.
The only corner-case is when it is set to the empty string. Though, the wording of the suggestion ("Please adjust...") seems clear enough to me.